src/functional/fencil_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -30,43 +34,53 @@ def get_param_description(
 ) -> source_modules.ScalarParameter | source_modules.BufferParameter:
     view = numpy.asarray(obj)
     if view.ndim > 0:
-        return source_modules.BufferParameter(name, [dim.value for dim in obj.axes], view.dtype)
+        return source_modules.BufferParameter(
+            name, tuple(dim.value for dim in obj.axes), view.dtype
+        )
     else:
         return source_modules.ScalarParameter(name, view.dtype)
 
 
-@fencil_source_module_generator
-def create_source_module(
-    itir: FencilDefinition,
-    *args,
-    **kwargs,
-) -> source_modules.SourceModule:
-    """Generate GTFN C++ code from the ITIR definition."""
-    parameters = [
-        get_param_description(itir_param.id, obj) for obj, itir_param in zip(args, itir.params)
-    ]
-    function = source_modules.Function(itir.id, parameters)
+@dataclass(frozen=True)
+class GTFNSourceModuleGenerator(FencilSourceModuleGenerator):
```

Comment:
This has become a class for the same reason as `gtfn_cpu.GTFNExecutor`. More detail there.